src/components/include/utils/threads/thread.h

```diff
@@ -90,7 +98,12 @@ void sleep(uint32_t ms);
 Thread* CreateThread(const char* name, ThreadDelegate* delegate);
 void DeleteThread(Thread* thread);
 
+#ifdef QT_PORT
+class Thread : public QObject {
```

Comment:
Could we avoid QObject inheritance?
As I see ``close_thread`` signal [DirectConnection connected](http://doc.qt.io/qt-5/qt.html#ConnectionType-enum)  and could be replaced with common method call.